CI: cuDF is failing

https://www.kaggle.com/code/marcogorelli/narwhals-pre-release-test-cudf?scriptVersionId=247959537

```
_________________________ ERROR collecting tests/get_dtype_backend_test.py _________________________
tests/get_dtype_backend_test.py:141: in <module>
    "pandas_dtype", generate_pandas_dtypes_public(SIMPLE_DTYPE_ALIASES), ids=str
tests/get_dtype_backend_test.py:134: in generate_pandas_dtypes_public
    return list(it)
tests/get_dtype_backend_test.py:123: in brute_force_construct_dtype
    result = dtype.construct_from_string(alias)
/usr/local/lib/python3.11/dist-packages/cudf/core/dtypes.py:319: in construct_from_string
    raise NotImplementedError()
E   NotImplementedError
===================================== short test summary info ======================================
ERROR tests/get_dtype_backend_test.py - NotImplementedError
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

@dangotbanned fancy taking a look if you have time please?

--- a/tests/get_dtype_backend_test.py
+++ b/tests/get_dtype_backend_test.py
@@ -119,7 +119,10 @@ def brute_force_construct_dtype(
 ) -> Iterator[ExtensionDtype]:
     """Skips dtypes that are not compatible on the tested version of pandas."""
     for dtype in dtypes:  # pragma: no cover
-        with suppress(TypeError):
+        # NOTE: Slightly different errors between implementations
+        # TypeError: https://github.com/pandas-dev/pandas/blob/2cc37625532045f4ac55b27176454bbbc9baf213/pandas/core/dtypes/base.py#L261-L264
+        # NotImplementedError: https://github.com/rapidsai/cudf/blob/22d1b7c75e892f9136e11e47c09e7913910414ea/python/cudf/cudf/core/dtypes.py#L318-L319
+        with suppress(TypeError, NotImplementedError):
             result = dtype.construct_from_string(alias)
             yield result
             return

